java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -47,23 +50,36 @@ public void start(Map<String, String> props) {
         context -> ofNullable(context.offsetStorageReader())).orElseThrow();
     AzureEventHubsConfig azureEventHubsConfig = new AzureEventHubsConfig(props);
     TopicPartitionOffsetProvider topicPartitionOffsetProvider = new TopicPartitionOffsetProvider(offsetStorageReader);
+    Kcql parsedKcql = Kcql.parse(
+        azureEventHubsConfig.getString(AzureEventHubsConfigConstants.KCQL_CONFIG).split(";")[0]);
+    KeyValueTypes keyValueTypes = getSourceDataTypes(parsedKcql);
 
-    ArrayBlockingQueue<ConsumerRecords<String, String>> recordsQueue = new ArrayBlockingQueue<>(
+    ArrayBlockingQueue<ConsumerRecords<Object, Object>> recordsQueue = new ArrayBlockingQueue<>(
         RECORDS_QUEUE_DEFAULT_SIZE);
     blockingQueueProducerProvider = new BlockingQueueProducerProvider(topicPartitionOffsetProvider);
     BlockingQueuedKafkaProducer producer = blockingQueueProducerProvider.createProducer(
-        azureEventHubsConfig, recordsQueue);
+        azureEventHubsConfig, recordsQueue, keyValueTypes);
     EventHubsKafkaConsumerController kafkaConsumerController = new EventHubsKafkaConsumerController(
         producer, recordsQueue);
     initialize(kafkaConsumerController, azureEventHubsConfig);
   }
 
+  private static KeyValueTypes getSourceDataTypes(Kcql parsedKcql) {
+    Map<String, String> kcqlProperties = parsedKcql.getProperties();
+    String keyFormat = kcqlProperties.getOrDefault("key.format", SourceDataType.BYTE.name());
+    String valueFormat = kcqlProperties.getOrDefault("value.format", SourceDataType.BYTE.name());
```

Comment:
The connector should have a standard way of exposing its capabilities with regards to its KCQL properties that are supported.  For the S3/Cloud connectors we have a KCQL properties "schema".  So if the user configures a property that doesn't exist, they will get a warning, and if the user configures a property with an invalid data type likewise.

I would like it if this this functionality could be pushed to the `kcql` lib that we're not reinventing the wheel, I don't think it would take much to reimplement it in Java and some improvements could be made along the way.

Reply:
Yeah, can be useful to print to users missing params or the ones that aren't even configured